web-ui/src/streaming/PipelineTable.tsx

```diff
@@ -0,0 +1,668 @@
+          .filter(ac => ac.direction === Direction.INPUT)
+          .map(ac => {
+            return { ac, c: connectorQuery.data.find(c => c.connector_id === ac.connector_id) }
+          })
+      )
+
+      setOutputs(
+        props.row.attached_connectors
+          .filter(ac => ac.direction === Direction.OUTPUT)
+          .map(ac => {
+            return { ac, c: connectorQuery.data.find(c => c.connector_id === ac.connector_id) }
+          })
+      )
+    }
+  }, [
+    connectorQuery.isLoading,
+    connectorQuery.isError,
+    connectorQuery.data,
+    setInputs,
+    setOutputs,
+    props.row.attached_connectors
+  ])
+
+  const [globalMetrics, setGlobalMetrics] = useState<GlobalMetrics[]>([])
+  const [inputMetrics, setInputMetrics] = useState<Map<string, InputConnectorMetrics>>(new Map())
+  const [outputMetrics, setOutputMetrics] = useState<Map<string, OutputConnectorMetrics>>(new Map())
+  const pipelineStatusQuery = useQuery<any>(['pipelineStatus', { pipeline_id: props.row.pipeline?.pipeline_id }], {
+    enabled: props.row.pipeline !== undefined && props.row.pipeline !== null,
+    refetchInterval: 1000
+  })
+
+  useEffect(() => {
+    if (!pipelineStatusQuery.isLoading && !pipelineStatusQuery.isError) {
+      console.log(pipelineStatusQuery.data)
+      const metrics = pipelineStatusQuery.data['global_metrics']
+      setGlobalMetrics(oldMetrics => [...oldMetrics, metrics])
+
+      const newInputMetrics = new Map<string, InputConnectorMetrics>()
+      pipelineStatusQuery.data['inputs'].forEach((cs: ConnectorStatus) => {
+        newInputMetrics.set(cs.endpoint_name, cs.metrics as InputConnectorMetrics)
+      })
+      setInputMetrics(newInputMetrics)
+
+      const newOutputMetrics = new Map<string, OutputConnectorMetrics>()
+      pipelineStatusQuery.data['outputs'].forEach((cs: ConnectorStatus) => {
+        newOutputMetrics.set(cs.endpoint_name, cs.metrics as OutputConnectorMetrics)
+      })
+      setOutputMetrics(newOutputMetrics)
+    }
+    if (props.row.pipeline == undefined) {
+      setGlobalMetrics([])
+      setInputMetrics(new Map())
+      setOutputMetrics(new Map())
+    }
+  }, [pipelineStatusQuery.isLoading, pipelineStatusQuery.isError, pipelineStatusQuery.data, props.row.pipeline])
+
+  return !connectorQuery.isLoading && !connectorQuery.isError && !projectQuery.isLoading && !projectQuery.isError ? (
+    <Box display='flex' sx={{ m: 2 }} justifyContent='center'>
+      <Grid container spacing={3} sx={{ height: 1, width: '95%' }} alignItems='stretch'>
+        <Grid item xs={4}>
+          <Card>
+            <List subheader={<ListSubheader>Configuration</ListSubheader>}>
+              <ListItem>
+                <ListItemIcon>
+                  <Icon icon='bi:filetype-sql' fontSize={20} />
+                </ListItemIcon>
+                <ListItemText primary={projectQuery.data?.name || 'not set'} />
+              </ListItem>
+              {props.row.pipeline && (
+                <>
```

Comment:
what is this?

Reply:
https://legacy.reactjs.org/docs/fragments.html#short-syntax